tests/BasicSearchTests.FunctionalTests.Core/V3SearchFunctionalTests.cs

```diff
@@ -99,11 +101,71 @@ public void ShouldThrowExceptionForInvalidSearchResults()
+        {
+            // Act
+            var response = await Client.GetAsync(new V3SearchBuilder() { Query = Constants.TestPackageId }.RequestUri);
+            var result = await response.Content.ReadAsAsync<V3SearchResult>();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.True(result.TotalHits.HasValue);
+            Assert.True(result.TotalHits.Value > 0, $"Could not find test package {Constants.TestPackageId}");
+        }
+
+        [Fact]
+        public async Task EnsureTestPackageIsValid()
+        {
+            // Act
+            var response = await Client.GetAsync(new V3SearchBuilder() { Query = Constants.TestPackageId }.RequestUri);
+            var result = await response.Content.ReadAsAsync<V3SearchResult>();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Assert.True(result.TotalHits.HasValue);
+            Assert.True(result.TotalHits.Value > 0, $"Could not find test package {Constants.TestPackageId}");
+            Assert.True(result.Data.Count > 0, $"Could not find test package {Constants.TestPackageId}");
+
+            Assert.NotNull(result.AtContext);
+            Assert.True(result.AtContext.AtVocab == "http://schema.nuget.org/schema#");
+            Assert.False(string.IsNullOrEmpty(result.AtContext.AtBase));
+            Assert.True(result.LastReopen.HasValue);
+            Assert.True(result.LastReopen.Value != default(DateTime));
+            Assert.False(string.IsNullOrEmpty(result.Index));
+
+            // Assert that the package result whose Id is "BaseTestPackage" with Version "1.0.0"
+            // matches exactly what is expected.
+            var package = result.Data
+                .Where(p => p.Id == Constants.TestPackageId)
+                .Where(p => p.Version == Constants.TestPackageVersion)
+                .FirstOrDefault();
+
+            Assert.NotNull(package);
+            Assert.False(string.IsNullOrEmpty(package.AtId));
+            Assert.True(package.AtType == "Package");
+            Assert.False(string.IsNullOrEmpty(package.Registration));
+            Assert.True(package.Description == "Package description");
+            Assert.True(package.Summary == "");
+            Assert.True(package.Title == "BaseTestPackage");
+            Assert.True(package.Tags.Count() == 2);
+            Assert.True(package.Tags[0] == "Tag1");
+            Assert.True(package.Tags[1] == "Tag2");
+            Assert.True(package.Authors.Count() == 1);
+            Assert.True(package.Authors[0] == "clayco");
+            Assert.True(package.TotalDownloads != default(long));
+            Assert.True(package.Versions.Count() == 1);
+            Assert.True(package.Versions[0].Version == "1.0.0");
+            Assert.True(package.Versions[0].Downloads != default(long));
+            Assert.False(string.IsNullOrEmpty(package.Versions[0].AtId));
```

Comment:
You could probably refactor these validations since reused above in V2 tests too. Up to you.